### PR TITLE
  feat (app): Add tron support

### PR DESCRIPTION
**Pull request by vrockz747**
Tron support for TRX transfer type transactions, txn decoding, signing and address generation

### CODE REVIEW

**Review by ParnikaGupta9 — CHANGES_REQUESTED** [review]
1. Add submodules for tron
2. Verify- public key, derivation path addresses and signature.

**Comment by vrockz747** [comment]
> 1. Add submodules for tron
> 
>     2. Verify- public key, derivation path addresses and signature.
> 
>     3. For sdk: get the current balance of the account and verify the signed transaction through tron testnet, then raise the PR.

submodules updated, verified address generation, signing and txn broadcasting

File: apps/tron_app/tron_helpers.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    return status;
+
+  uint32_t purpose = path[0], coin = path[1], account = path[2];
+
+  switch (levels) {
+    case 3: {    // m/44'/195'/0'
+      status = (purpose == TRON_PURPOSE_INDEX && coin == TRON_COIN_INDEX &&
+                account == TRON_ACCOUNT_INDEX);
+    } break;
+
```
Comment (VeronicaGupta) [inline apps/tron_app/tron_helpers.c:109]: Does case 4 exist? Include the reference for this.

Comment (vrockz747) [reply apps/tron_app/tron_helpers.c:109]: 1st one (m/44'/195'/0') is tron's standard path, 
second one (m/44'/195'/0'/0/i) is popular path used by adopted by other wallets

File: apps/tron_app/tron_txn_helpers.c [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  if (!pb_decode(&stream, TRON_TRANSACTION_RAW_FIELDS, raw_txn)) {
+    return -1;
+  }
+
+  return 0;
+}
+
+int tron_validate_unsigned_txn(const tron_transaction_raw_t *raw_txn) {
+  // TODO:
+  return 0;
```
Comment (ParnikaGupta9) [inline apps/tron_app/tron_txn_helpers.c:119]: add validation like:
if contract count is non-zero
if fee limit can be non zero
if contract is not null

File: apps/tron_app/tron_txn_helpers.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+#define TRON_NAME "Tron"
+#define TRON_LUNIT "TRX"
+#define TRON_PUB_KEY_SIZE 64    // ECDSA's x+y coordinates only
+#define TRON_PUB_KEY_SIZE_PLUS_ONE 65
+#define TRON_INITIAL_ADDRESS_LENGTH 21
+#define TRON_INITIAL_ADDRESS_LENGTH_MINUS_ONE 20
+#define TRON_ACCOUNT_ADDRESS_LENGTH 34
```
[the remark this answers is not in the document]
Comment (vrockz747) [reply apps/tron_app/tron_txn_helpers.h:33]: updated

File: apps/tron_app/tron_helpers.c (hunk 1 above)
Comment (vrockz747) [reply apps/tron_app/tron_helpers.c:109]: updated

File: apps/tron_app/tron_txn_helpers.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (vrockz747) [reply apps/tron_app/tron_txn_helpers.h:33]: updated

File: apps/tron_app/tron_contracts.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+// a9059cbb -> transfer(address,uint256)
+const uint8_t selector[TRC20_FUNCTION_SELECTOR_COUNT][4] = {
+    {0xA9, 0x05, 0x9C, 0xBB}};
+
+const trc20_contracts_t trc20_contracts[TRC20_WHITELISTED_CONTRACTS_COUNT] = {
+    {{0x41, 0xA6, 0x14, 0xF8, 0x03, 0xB6, 0xFD, 0x78, 0x09, 0x86, 0xA4,
+      0x2C, 0x78, 0xEC, 0x9C, 0x7F, 0x77, 0xE6, 0xDE, 0xD1, 0x3C},
+     "USDT",
+     6},
+    {{0x41, 0x34, 0x87, 0xb6, 0x3d, 0x30, 0xb5, 0xb2, 0xc8, 0x7f, 0xb7,
```
[the remark this answers is not in the document]
Comment (vrockz747) [reply apps/tron_app/tron_contracts.c:98]: added

File: apps/tron_app/tron_contracts.c [hunk 5]
```diff
@@ -0,0 +1,138 @@
+/**
+ * @file    TRON_contracts.c
+ * @author  Cypherock X1 Team
```
[the remark this answers is not in the document]
Comment (vrockz747) [reply apps/tron_app/tron_contracts.c:3]: fixed

File: apps/tron_app/tron_contracts.c [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    {{0x41, 0x74, 0x47, 0x2E, 0x7D, 0x35, 0x39, 0x5A, 0x6B, 0x5A, 0xDD,
+      0x42, 0x7E, 0xEC, 0xB7, 0xF4, 0xB6, 0x2A, 0xD2, 0xB0, 0x71},
+     "WIN",
+     6},
+    {{0x41, 0x89, 0x1c, 0xdb, 0x91, 0xd1, 0x49, 0xf2, 0x3b, 0x1a, 0x45,
+      0xd9, 0xc5, 0xca, 0x78, 0xa8, 0x8d, 0x0c, 0xb4, 0x4c, 0x18},
+     "WTRX",
+     6},
+};
+
```
[the remark this answers is not in the document]
Comment (vrockz747) [reply apps/tron_app/tron_contracts.c:131]: added

File: apps/tron_app/tron_helpers.h [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+#include "coin_utils.h"
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+#define TRON_PURPOSE_INDEX 0x8000002C
+#define TRON_COIN_INDEX 0x800000C3
+#define TRON_ACCOUNT_INDEX 0x80000000
+#define TRON_CHANGE_INDEX 0x0
+/*****************************************************************************
```
[the remark this answers is not in the document]
Comment (vrockz747) [reply apps/tron_app/tron_helpers.h:29]: updated

File: apps/tron_app/tron_txn_helpers.c (hunk 2 above)
Comment (vrockz747) [reply apps/tron_app/tron_txn_helpers.c:119]: added

**Pull request closed without merging**